### DISCUSSION TITLE
newer versions use {c:\users\"user"\.hfs} folder instead of program directory where hfs.exe lives.

### DISCUSSION THREAD

**Discussion by badmuts71** [rejetto/hfs#758] [Q&A]
Is there no more portable version of HFS? Until upto version 0.54-alpha1 config files were stored in program directory.
When i update now to newer version , all settings , certificate and shared sources are gone...and fresh config files appear to be on C drive..

**Comment by rejetto** [comment]
hfs.exe is supposed to store files in same directory by default.
It is possible that a bug in latest version stopped this behavior, I will check tomorrow.
You can force the folder to be the same by entering the folder with cmd and running `hfs --cwd .`
because "." is the the current folder
If you run it this way, you will see your settings back in place.
There's no need for a "portable version" because you can decide where files are stored.

**Comment by rejetto** [comment]
i just tried to run hfs on my Windows, and it's working correctly
i have it in C:\hfs\ and when i run it, it says "working directory C:\hfs"
it's not trying to use "users" folder.
Where do you keep your hfs? And how do you launch it?

**Comment by badmuts71** [comment]
Thanks for taking time to investigate!

I downloaded a new version of  HFS version 0.54.0-alpha4 on another PC and indeed settings are in working directory.

Started HFS on PC that works with version 0.54.0-alpha1 and this is shown in command window:
"20:50:12 cwd C:\Program Files Portable\hfs-windows"

Then i updated that version with latest, HFS closes and starts again , and this is shown in command window:
"20:56:51 working directory C:\Users\Administrator\.hfs"

"cwd" was changed to "working directory" tekstual (could be due to your changes in coding?) ,  settings are lost and new config is indeed in:
C:\Users\Administrator\.hfs

**Comment by badmuts71** [comment]
i forgot to mention that your solution of forcing it with: "hfs --cwd ." does work, but updating shortcut to:
C:\Program Files Portable\hfs-windows\hfs --cwd . and C:\Program Files Portable\hfs-windows\hfs.exe --cwd . fails...
Might be a spacing issue of folder structure.

Moved HFS to c:\HFS \ as you have and everything works fine, i'll just keep it there, case closed...

**Comment by rejetto** [comment]
wait, "--cwd ." can solve at command line, but if you have a "shortcut" the cwd is the field "starts in". 
<img width="179" alt="image" src="https://github.com/user-attachments/assets/1a16075c-a8b5-413b-b4c8-8fc673c57951">

You could fill that field with "C:\Program Files Portable\hfs-windows", and that may solve the problem.
but i'm saying just FYI, because the problem is actually another 😄
the reason you didn't have the problem with alpha1 is because i recently made a change.
When HFS detects it is running in "program files" folder, it automatically uses the "users" folder to save.
I made this change after [a user reported](https://github.com/rejetto/hfs/discussions/715#discussioncomment-10438177) that HFS running in "program files" couldn't save settings.
And that was quite my experience, that programs are not allowed to write to "program files".
So, did I do wrong?

**Comment by rejetto** [comment]
even if considering "program files" not writable is correct,
HFS is currently confusing your "program files portable" folder with "program files".
it's actually a bug, and it will be fixed for next release.
if it wans't for this bug, you wouldn't have had this accident.
